index.py

```diff
@@ -27,24 +27,44 @@ def restore_index(self):
             logging.info('No previous index was found')
 
     def search(self, query: str, number_of_results: int):
-        if len(query.split(' ')) > 1:
-            raise AttributeError('Only single word query\'s are supported for now')
         processed_query = self.get_tokens(query)
-        results = list()
+
+        result_stack = []
+        pending_operation = ''
         for token in processed_query:
-            pointers = self.index.get(token)
-            if not pointers:
-                continue
-            i = 0
-            for pointer in pointers:
-
-                comment = CsvHelper.read_comment(pointer)
-                results.append(comment)
-                if i > number_of_results:
-                    break
-                i += 1
-
-        return results
+            if token == 'and' or token == 'or' or token == 'not':
+                pending_operation = token.upper()
+            else:
+                results = set()
+                pointers = self.index.get(token)
+                if not pointers:
+                    continue
+                i = 0
+                for pointer in pointers:
+                    comment = CsvHelper.read_comment(pointer)
+                    results.add(comment)
+                    if i > number_of_results:
+                        break
+                    i += 1
+                result_stack.append(results)
+
+                if pending_operation != '':
+                    results = getattr(self, pending_operation)(result_stack.pop(), result_stack.pop())
```

Comment:
Isn't `result_stack.pop()` returning two times the same?

Reply:
nope. it deletes the recieved element from the list.